src/main/java/org/jenkinsci/plugins/edithelp/EditHelpRootAction.java

```diff
@@ -0,0 +1,156 @@
+package org.jenkinsci.plugins.edithelp;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.BufferedWriter;
+import java.io.OutputStreamWriter;
+import java.io.FileOutputStream;
+import hudson.Extension;
+import hudson.cli.PrivateKeyProvider;
+import hudson.model.RootAction;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.QueryParameter;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import hudson.util.HttpResponses;
+
+import static java.nio.file.Files.readAllLines;
+import static java.util.logging.Level.WARNING;
+
+
+
+
+
+@Extension
+public class EditHelpRootAction implements RootAction {
+
+    private static final Logger LOGGER = Logger.getLogger(PrivateKeyProvider.class.getName());
+
+    //hashMap for cache storage
+    private transient Map<String, String> hashMapOfHelp = new HashMap<String, String>();
+
+    //reading files into cache
+    public EditHelpRootAction() throws IOException, FileNotFoundException {
+        File dirfile = Jenkins.getActiveInstance().getRootDir();
+        if(dirfile != null) {
+
+            //check if the directory exists
+            File helpManagerFile = new File(dirfile, "helpmanager");
+
+            if (!helpManagerFile.exists()) {
+                if (!helpManagerFile.mkdirs()) {
+                    LOGGER.log(WARNING,"Unable create directory: "+helpManagerFile.getAbsolutePath());
+                }
+            }
+
+            //reading files from the directory
+            if (helpManagerFile.getAbsolutePath() != null) {
+                File dir = new File(helpManagerFile.getAbsolutePath());
+                File[] files = dir.listFiles();
+                if (files != null) {
+                    for (File file : files){
+                        if (file.isFile()){
+                            hashMapOfHelp.put(file.getName(),readFromFile(file));
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public String getIconFileName() {
+        return null;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return null;
+    }
+
+    @Override
+    public String getUrlName() {
+        return "helpmanager";
+    }
+
+    //copying from HashMap
+    @Restricted(NoExternalUse.class)
+    public HttpResponse doHelpInfo(@QueryParameter("class") String className) {
+
+        Jenkins.getActiveInstance().checkPermission(Jenkins.READ);
```

Comment:
This solution is far from perfect, but it is better than nothing.
Ideally you should retrieve AccessControlled items from ancestor path and check permissions for them